src/loops.rs

```diff
@@ -228,6 +231,17 @@ impl LateLintPass for LoopsPass {
                 }
             }
         }
+        if let ExprMatch(ref expr, ref arms, MatchSource::WhileLetDesugar) = expr.node {
+            let pat = &arms[0].pats[0].node;
+            if let (&PatEnum(ref path, _), &ExprMethodCall(method_name, _, _)) = (pat, &expr.node) {
+                if method_name.node.as_str() == "next" &&
+                        match_trait_method(cx, expr, &["core", "iter", "Iterator"]) &&
+                        path.segments.last().unwrap().identifier.name.as_str() == "Some" {
```

Comment:
I can do that if you prefer, but let me defend `unwrap` here for a second: As far as I can tell, `path.segments` should always be non-empty, so `last` should always return a `Some`. Now, I might be wrong about that, but if I am I'd rather have the program crash instead of silently carry on in some state that I didn't anticipate. (Ok, I should still use `expect` so clippy at least crashes with a sensible error message.)